tedana/reporting/dynamic_figures.py

```diff
@@ -0,0 +1,384 @@
+                          tools=["tap,wheel_zoom,reset,pan,crosshair", hovertool],
+                          title=title)
+    fig.circle(x_var, y_var, source=comptable_ds, size=3, color='color')
+    fig.xaxis.axis_label = x_label
+    fig.yaxis.axis_label = y_label
+    fig.x_range = models.Range1d(-1, n_comps + 1)
+    fig.toolbar.logo = None
+
+    return fig
+
+
+def tap_callback(comptable_ds, ts_src, fft_src,
+                 ts_plot, fft_plot,
+                 div_content, out_dir):
+    """
+    Javacript function to animate tap events and show component info on the right
+
+    Parameters
+    ----------
+    comptable_ds: bokeh.models.ColumnDataSource
+        Data structure containing a limited set of columns from the comp_table
+    ts_src :
+    fft_src :
+    ts_plot :
+    fft_plot :
+    div : bokeh.models.Div
+        Target Div element where component images will be loaded
+
+    Returns
+    -------
+    CustomJS: bokeh.models.CustomJS
+        Javascript function that adds the tapping functionality
+    """
+    return models.CustomJS(
+        args=dict(source_comp_table=comptable_ds,
+                  source_meica_ts=ts_src,
+                  source_tsplot=ts_plot,
+                  source_meica_fft=fft_src,
+                  source_fftplot=fft_plot,
+                  div=div_content, outdir=out_dir),
+        code="""
+    // Accessing the selected component ID
+    var data     = source_comp_table.data;
+    var selected = source_comp_table.selected.indices;
+    var selected_padded = '' + selected;
+    while (selected_padded.length < 3) {
+        selected_padded = '0' + selected_padded;
+    }
+    // Creating a new version 000 --> C000
+    var selected_padded_C = 'C' + selected_padded
+
+    // Updating TS Plot
+    var Plot_TS = source_tsplot.data;
+    var TS_x    = Plot_TS['x']
+    var TS_y    = Plot_TS['y']
+    var Comp_TS   = source_meica_ts.data;
+    var Comp_TS_y = Comp_TS[selected_padded_C]
+
+    for (var i = 0; i < TS_x.length; i ++) {
+        TS_y[i] = Comp_TS_y[i]
+    }
+    source_tsplot.change.emit();
+
+    // Updating FFT Plot
+    var Plot_FFT = source_fftplot.data;
+    var FFT_x = Plot_FFT['x']
+    var FFT_y = Plot_FFT['y']
+    var Comp_FFT = source_meica_fft.data;
+    var Comp_FFT_y = Comp_FFT[selected_padded_C]
+    for (var i = 0; i < FFT_x.length; i ++) {
+        FFT_y[i] = Comp_FFT_y[i]
+    }
+    source_fftplot.change.emit();
+
+
+    console.log('selected = ' + selected_padded)
+    div.text = ""
+    var line = "<span><img src='" + outdir + "/figures/comp_"+selected_padded+".png'" +
+        " alt='Component Map' height=1000 width=900><span>\\n";
+    console.log('Linea: ' + line)
+    var text = div.text.concat(line);
+    var lines = text.split("\\n")
+        if (lines.length > 35)
+            lines.shift();
+    div.text = lines.join("\\n");
+    """)
+
+
+# %%
+# ----------------------------------------------------------------
```

Comment:
This section was me trying to get something to quickly test and show. we should remove everything below L335, here.

Reply:
@emdupre: I think I'm a little confused -- we don't want both the KappaRho_DynPlot.py file and the dynamic_figures.py file. If KappaRho_DynPlot is intended to supersede dynamic_figures, can we remove the current dynamic_figures file and rename KappaRho_DynPlot as "dynamic_figures" ? Correct, KappaRho_DynPlot supersedes dynamic_figures